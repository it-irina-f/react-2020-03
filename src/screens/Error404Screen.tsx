import React from "react";
import styled from "@emotion/styled";

const Container = styled.div`
  margin: 40px auto;
  text-align: center;
  padding: 10px;
  font-size: 200%;
`;

export const Error404Screen: React.FC<{}> = () => {
  return (
    <Container>
      <h1>404</h1>
      <p>Page not found</p>
    </Container>
  );
};
