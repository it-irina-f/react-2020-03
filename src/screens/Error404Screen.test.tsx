import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { Error404Screen } from "./Error404Screen";

describe("Component Error404Screen", () => {
  it("renders page Error 404", () => {
    expect(renderer.create(<Error404Screen />).toJSON()).toMatchSnapshot();
  });

  const screen = shallow(<Error404Screen />);

  it("renders h1 and p", () => {
    expect(screen.find("h1").text()).toEqual("404");
    expect(screen.find("p").text()).toEqual("Page not found");
  });
});
