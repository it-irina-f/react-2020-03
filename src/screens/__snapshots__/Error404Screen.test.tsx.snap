// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Error404Screen renders page Error 404 1`] = `
.emotion-0 {
  margin: 40px auto;
  text-align: center;
  padding: 10px;
  font-size: 200%;
}

<div
  className="emotion-0"
>
  <h1>
    404
  </h1>
  <p>
    Page not found
  </p>
</div>
`;
